include/sched_priq.h

```diff
@@ -56,8 +56,8 @@ struct _priq_mq {
 	unsigned int bitmask; /* bit 1<<i set if queues[i] is non-empty */
 };
 
-void _priq_mq_add(struct _priq_mq *pq, struct k_thread *thread);
-void _priq_mq_remove(struct _priq_mq *pq, struct k_thread *thread);
+ALWAYS_INLINE void _priq_mq_add(struct _priq_mq *pq, struct k_thread *thread);
```

Comment:
this doesn't look correct

what does ALWAYS_INLINE mean for functions that are non-static and implemented elsewhere?

Reply:
I guess thats why sanitycheck is failing almost on everything, @dcpleung did you try to build with !xcc?

Reply:
I think the entire function definition needs to be moved to these header files if they are intended to always be inlined

Reply:
I built with qemu_x86 a few days back but with an older patch. Let me take a look.

Reply:
Probably better to move the declarations internal to sched.c and eliminate the header entirely.  It's vestigial, from a time when the scheduler API was more diffuse and these needed to be seen elsewhere.